### PR TITLE
doc: ArgoCD + FluxCD App Listing

### PR DESCRIPTION
**Pull request by ashokdevtron**
- ArgoCD App and FluxCD App added in Glossary
- Edited **applications.md** and expanded it to include Argo CD app listing and Flux CD app listing
- Created a video tutorial to enable 'ArgoCD Apps' tab.
- Snapshot added
- RBAC added

### CODE REVIEW

**Review by abhibhaw — CHANGES_REQUESTED** [review]
Needs minor restructuring and messaging

File: docs/user-guide/applications.md [hunk 1]
```diff
@@ -1,3 +1,133 @@
...
+## View FluxCD App Listing
+
+{% hint style="warning" %}
+### Who Can Perform This Action?
+Users need super-admin permission to view/enable/disable the FluxCD listing.
+{% endhint %}
+
+### Preface
+
+Devtron also supports the listing of [Flux CD](https://fluxcd.io/) apps in one dashboard, thus providing the same [advantages](#advantages) of [ArgoCD app listing](#view-argocd-app-listing).
```
Comment (abhibhaw) [inline docs/user-guide/applications.md:111]: `of argocd` or `as argocd`?

File: docs/user-guide/applications.md [hunk 2]
```diff
@@ -1,3 +1,133 @@
...
+
+### Preface
+
+In Argo CD, a user manages one dashboard for each Argo CD app. Therefore, with multiple apps, the process becomes cumbersome for the user to manage several dashboards.
+
+With Devtron, you get an entire Argo CD app listing in one place. This listing includes:
+* Apps deployed using [GitOps](../reference/glossary.md#gitops) on Devtron
+* Other Argo CD apps present in your cluster
+
+![Figure 2: ArgoCD App List](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/argocd/argo-cd-listing.jpg)
```
Comment (abhibhaw) [inline docs/user-guide/applications.md:52]: we should have an screenshot of argo app details

File: docs/user-guide/applications.md [hunk 3]
```diff
@@ -1,3 +1,133 @@
...
+The cluster in which Flux CD apps exist should be added in **Global Configurations** → **Clusters and Environments**
+
+### Enabling FluxCD App Listing
+
+{% hint style="info" %}
+### Tip
+You may refer the steps mentioned in the [Enabling ArgoCD App Listing](#enabling-argocd-app-listing) section since the procedure is similar. 
+{% endhint %}
+
+The feature flag to enable FluxCD app listing is **FEATURE_EXTERNAL_FLUX_CD_ENABLE** and it should be set to **"true"** as shown below.
```
Comment (abhibhaw) [inline docs/user-guide/applications.md:125]: It should be like, here is the feature flag, refer how to add a feature flag in dashboard / orchestrator....

File: docs/user-guide/applications.md [hunk 4]
```diff
@@ -1,3 +1,133 @@
...
+
+![Figure 5: Editing Dashboard ConfigMap](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/fluxcd/flux-feature-flag.jpg)
+
+After successfully executing all the steps, a new tab named **FluxCD Apps** will be visible. Select the cluster(s) from the dropdown to view the Flux CD apps available in the chosen cluster(s).
```
Comment (abhibhaw) [inline docs/user-guide/applications.md:129]: A flux app detail screenshot should be added

**Review by abhibhaw — CHANGES_REQUESTED** [review]

File: docs/user-guide/applications.md [hunk 5]
```diff
@@ -1,3 +1,133 @@
...
+
+The **Applications** page helps you create and manage your microservices, and it majorly consists of the following:
+
+* [Application Listing](#application-listing)
+* [Create Button](#create-button)
+* [Other Options](#other-options)
+
+### Application Listing
+
+You can view the app name, its status, environment, namespace, and many more upfront. The apps are segregated into: [Devtron Apps](../reference/glossary.md#devtron-apps), [Helm Apps](../reference/glossary.md#helm-apps), and [ArgoCD Apps](../reference/glossary.md#argocd-apps)
```
[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/user-guide/applications.md]: Hmm, yes. Added.

File: docs/reference/glossary.md [hunk 6]
```diff
@@ -10,6 +10,12 @@ An immutable blob of data generated as an output after the execution of a job, b
 
 * Once a job is complete, you can view the job artifacts by going to Jobs → Run history (tab) → (choose a pipeline and date of triggering the build) → Artifacts (tab).
 
+### ArgoCD Apps
```
[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/reference/glossary.md:13]: Added

File: docs/user-guide/applications.md (hunk 4 above)
Comment (ashokdevtron) [reply docs/user-guide/applications.md:137]: Added a gif for the better understanding of our readers

File: docs/user-guide/applications.md [hunk 7]
```diff
@@ -1,3 +1,145 @@
...
+## View ArgoCD App Listing
+
+{% hint style="warning" %}
+### Who Can Perform This Action?
+Users need super-admin permission to view/enable/disable the ArgoCD listing.
+{% endhint %}
+
+### Preface
+
+In Argo CD, a user manages one dashboard for each Argo CD app. Therefore, with multiple apps, the process becomes cumbersome for the user to manage several dashboards.
```
[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/user-guide/applications.md]: Done

File: docs/user-guide/applications.md [hunk 8]
```diff
@@ -1,3 +1,145 @@
...
+## View FluxCD App Listing
+
+{% hint style="warning" %}
+### Who Can Perform This Action?
+Users need super-admin permission to view/enable/disable the FluxCD listing.
+{% endhint %}
+
+### Preface
+
+Devtron also supports the listing of [Flux CD](https://fluxcd.io/) apps in one dashboard, thus providing the same [advantages](#advantages) as [ArgoCD app listing](#view-argocd-app-listing).
```
[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/user-guide/applications.md]: Added

File: docs/user-guide/applications.md [hunk 9]
```diff
@@ -1,3 +1,145 @@
...
+* Apps deployed using [GitOps](../reference/glossary.md#gitops) on Devtron
+* Other Argo CD apps present in your cluster
+
+![Figure 2: ArgoCD App List](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/argocd/app-details-argo.gif)
+
+### Advantages
+
+Devtron also bridges the gap for ArgoCD users by providing additional features as follows:
+
+* **Resource Scanning**: You can scan for vulnerabilities using Devtron's [resource scanning](../user-guide/security-features.md#from-app-details) feature.
```
[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/user-guide/applications.md]: Added enterprise tagging

**Review by abhibhaw — APPROVED** [review]

**Review by uxarya-d — APPROVED** [review]

**Pull request merged**